shared/js/sk_key_utils.js

```diff
@@ -0,0 +1,367 @@
+          header.textContent = options.header || '';
+        }
+
+        form.appendChild(header);
+      }
+
+      // We append a menu with the list of options
+      var menu = document.createElement('menu');
+      menu.dataset.items = items.length;
+
+      // For each option, we append the item and listener
+      items.forEach(function renderOption(item) {
+        var button = document.createElement('button');
+        button.type = 'button';
+        if (item.label && item.label.length) {
+          button.textContent = item.label || '';
+        } else {
+          return;
+        }
+
+        if (item.icon) {
+          button.style.backgroundImage = 'url("' + item.icon + '")';
+          button.classList.add('icon');
+        }
+
+        // Listener will be removed once we clean up the DOM
+        button.addEventListener('click', function (event) {
+          var clickEvt = new MouseEvent('click', {
+            'view': window,
+            'bubbles': true,
+            'cancelable': true
+          });
+          item.dispatchEvent(clickEvt);
+        });
+
+        menu.appendChild(button);
+      });
+
+      // Appending the action menu to the form
+      form.appendChild(menu);
+
+      // Add cancel key
+      var footer = document.createElement('footer');
+      footer.textContent = sk_options_menu.cancelKeyText;
+      footer.classList.add('sk-cancel-key');
+
+      // TODO: Need to change l10nid, wrong way
+      footer.setAttribute('data-l10n-id',
+              sk_options_menu.cancelKeyl10nId);
+
+      // Handle Close click, #IFDEF check required
+      // for TOUCH UI
+      footer.addEventListener('click', function () {
+        hideOptionsMenu();
+      });
+
+      form.appendChild(footer);
+
+      // Show Menu
+      sk_container.insertBefore(form, sk_container.firstChild);
+      form.classList.add('visible');
+
+      sk_options_menu.open = true;
+    }
+
+    /**
+     * Used for hiding options menu
+     * @example
+     * hideOptionsMenu();
+     *
+     * @param N/A
+     * @return void 0
+     */
+    function hideOptionsMenu() {
+      sk_container
+              .querySelector('form')
+              .remove('visible');
+
+      sk_options_menu.open = false;
+    }
+
+    /**
+     * Used to observe DOM changes
+     * @example
+     * observeDOMChange(element);
+     *
+     * @param {object} target: Target element on which observer will observe
+     * @return void 0
+     */
+    function observeDOMChange(target) {
+      var MutationObserver = window.MutationObserver ||
+              window.WebKitMutationObserver ||
+              window.MozMutationObserver;
+      if (!MutationObserver) {
+        throw Error('MutationObserver not supported');
+      }
+
+      observer = new MutationObserver(function (mutations) {
+        // TODO: Do we need to initialize again or handle each change.
+        //       Handling each change might be difficult
+        // init();
+
```

Comment:
You shouldn't have to re-initialize; that would be too slow.  Think about the kinds of changes we care about:

1) if the app changes the textContent of any of the buttons with accesskey on them, then we need to mirror that change in the soft key labels.  You can do that by creating one MutationObserver for each of those buttons (never more than 3 of them).  Those observers only have to pay attention to characterdata changes, I think.

2) You also need to know if accesskey attributes are added or removed anywhere in the document. And if that happens, you'll need to change the appropriate soft key.

3) You probably also have to monitor changes to the document itself, to see whether the elements with accesskey set are removed from the document.  Or maybe you can just monitor the 1, 2, or 3 current elements directly (observing the child list of the parent, maybe?)

4) You need to know if a new element with an access key attribute is added. You may not need to do anything for this: it could be that you'll catch those things just by monitoring attribute changes.

You'll want to be sure to not start monitoring until after the DOMContentLoaded event or after the load event. Otherwise your observers will be triggered all the time as the document is being built. If you use <script deferred> then that will probably be fine